navlie/utils.py

```diff
@@ -885,19 +886,31 @@ def plot_poses(
     # TODO. handle 2D case
     if isinstance(poses, GaussianResultList):
         poses = poses.state
-    
+
     if isinstance(poses, StateWithCovariance):
         poses = [poses.state]
-    
+
     if isinstance(poses, np.ndarray):
         poses = poses.tolist()
-        
+
     if not isinstance(poses, list):
         poses = [poses]
- 
+
+    # Check if poses are in 2D or 3D
+    if poses[0].position.shape[0] == 2:
```

Comment:
probably slightly better to do `position.size == 2` just to protect against [2 x N] matrices?